src/getMigrateDashboards.ts

```diff
@@ -61,101 +70,120 @@ export const getMigrateDashboards =
     const filesAncestry = _getFilesAncestry(structure);
 
     for (const fileId in content.children) {
-      if (!filesAncestry[fileId]) {
-        counters.dashboards.removed++;
-        _addCorruptFileErrorToReport(errorReport, {
-          contentType: "dashboards",
-          fileId,
-        });
-        continue;
-      }
+      if (idsOfDashboardsToMigrate.has(fileId)) {
```

Comment:
If we go this route, we could go `for fileId of idsOfDashboardsToMigrate` directly.

But if I recall the design session correctly, we had agreed on another design, based on `failedDashboardIds`.
I find it a bit more descriptive/intuitive (as it describes the extra piece of information that we need explicitly rather than implicitly; for example it would make it easier to write unit tests), and it should always be smaller than `idsOfDashboardsToMigrate` (which in the best case scenario is always all the ids).

With the same idea, we could even decide not to introduce a new `failedDashboardIds` variable, since we essentially already have the `errorReport` to store that  information.
If we go for that route we'd need to change things a little bit though if we want to keep the functional design of "always keep going when the migration of a widget within a dashboard fails".

If you had decided to go for `idsToMigrate` on purpose (vs `failedIds`), can you explain why you think it's better ?